src/solvers/td3.ml

```diff
@@ -651,44 +668,43 @@ module WP =
                 HM.replace old_ret k (old_rho, old_infl)
               )
             ) obsolete_ret;
-        ) else (
-          (* If reluctant destabilization is turned off we need to destabilize all nodes in completely changed functions
-             and the primary obsolete nodes of partly changed functions *)
-          print_endline "Destabilizing changed functions and primary old nodes ...";
-          HM.iter (fun k _ ->
-              if HM.mem stable k then
-                destabilize k
-            ) obsolete_entry;
-          HM.iter (fun k _ ->
-              if HM.mem stable k then
-                destabilize k
-            ) obsolete_prim;
         );
 
+        if not (HM.is_empty obsolete_entry) || not (HM.is_empty obsolete_prim) then
+          print_endline "Destabilizing changed functions and primary old nodes ...";
+        HM.iter (fun k _ ->
+            if HM.mem stable k then
+              destabilize k
+          ) obsolete_entry;
+        HM.iter (fun k _ ->
+            if HM.mem stable k then
+              destabilize k
+          ) obsolete_prim;
```

Comment:
This will probably cause problems when `reluctant` and CFG-comparison are activated.

Reply:
Should be resolved with only collecting `obsolete_prim` when `reluctant` is off.